x/crosschain/keeper/grpc_query_cctx.go

```diff
@@ -177,3 +178,260 @@ func (k Keeper) CctxListPending(c context.Context, req *types.QueryListCctxPendi
 		TotalPending: totalPending,
 	}, nil
 }
+
+// CctxListPendingWithinRateLimit returns a list of pending cctxs that do not exceed the outbound rate limit
+// a limit for the number of cctxs to return can be specified or the default is MaxPendingCctxs
+func (k Keeper) CctxListPendingWithinRateLimit(c context.Context, req *types.QueryListCctxPendingWithRateLimitRequest) (*types.QueryListCctxPendingWithRateLimitResponse, error) {
```

Comment:
Can we create a new file for the query?
`grpc_query_cctx_rate_limit.go`

Reply:
yeah.

Reply:
https://github.com/zeta-chain/node/pull/2045/commits/f31c988378329b4e88a85ebeee43b57990b09b92